### ISSUE TITLE
Staticcheck ST1003

### ISSUE THREAD

**Issue by roychoo** [golangci/golangci-lint#649]
Hi,
this is not a bug report.

want to ask regarding the configuration of staticcheck, i.e ST1003 as some check are not enabled by default, i tried to find more info in the documents but to no avail. It doesnt seem to support ```staticcheck.conf```

Please do let me know how can i configure it

**Comment by terwey** [comment]
I managed to get this working by adding:

```
issues:
  # Excluding configuration per-path, per-linter, per-text and per-source
  exclude-rules:
    - linters:
       - stylecheck
      text: "ST1003:"

    - linters:
       - stylecheck
      text: "ST1000:"
```

**Comment by ernado** [comment]
I think that the initial issue was about enabling back default excludes.
This will be possible via `include` option in upcoming release.

**Comment by csilvers** [comment]
It supported staticcheck.conf for me!  Perhaps this has been fixed in the latest golangci-lint.

**Comment by jbelmont** [comment]
How did you get it to work with config file?

**Comment by csilvers** [comment]
We just put a staticcheck.conf in our repo-root and ran golangci-lint, and it respected the conf file.  I don't remember the details now, unfortunately.

**Comment by jbelmont** [comment]
So we have the following *staticcheck.conf* file:

```toml
checks = ["all", "-SA6005", "-SA9003", "-SA4010", "-SA4006"]
```

Here is our *.golangci.yml* file:

```yml
run:
  deadline: 5m

linters-settings:
  gofmt:
    simplify: false

  govet:
    check-shadowing: false

    enable:
      - unmarshal
      - printf
      - tests
      - bools
      - assign
      - buildtag
      - unreachable
    enable-all: false
    disable-all: true

linters:
  enable:
    - gofmt
    - govet
    - staticcheck

  enable-all: false
  disable-all: true
  fast: false

issues:
  max-issues-per-linter: 50
  max-same-issues: 10

  exclude-rules:
    # Exclude some linters from running on tests files.
    - path: _test\.go
      linters:
        - gofmt
```

When we run the following command:

```bash
golangci-lint run ./...
```

We are seeing entries still for *SA6005* and *SA4010* so it doesn't seem to respect the staticcheck conf file

We tried the following version 1.30 and 1.27:

```bash
golangci-lint --version
golangci-lint has version 1.30.0 built from 45b90f6 on 2020-08-03T03:10:34Z
```

Is there anything we are doing wrong that you see?

**Comment by csilvers** [comment]
Not that I can see, but I'm no expert on staticcheck for sure.

Our staticcheck.conf file looked like this:
```
initialisms = [
    # These are the defaults from https://staticcheck.io/docs/options
    "ACL", "API", "ASCII", "CPU", "CSS", "DNS", "EOF", "GUID", "HTML",
    "HTTP", "HTTPS", "ID", "IP", "JSON", "QPS", "RAM", "RPC", "SLA",
    "SMTP", "SQL", "SSH", "TCP", "TLS", "TTL", "UDP", "UI", "GID",
    "UID", "UUID", "URI", "URL", "UTF8", "VM", "XML", "XMPP", "XSRF",
    "XSS",
    # Google initialisms.
    "GCS", "GCE", "GCP", "GAE",
]
```
So maybe only `initialisms` is supported?  You could try that and see if at least that kind of modification works.

**Comment by jbelmont** [comment]
Sorry for the late reply @csilvers we ended up just doing this:

![Screen Shot 2020-08-18 at 11 18 09 AM](https://user-images.githubusercontent.com/2974744/90531858-8eeace00-e144-11ea-85b3-b1f29612a9f2.png)

Essentially adding a yaml list per static check to disable, this was the only reliable way to get this working since we couldn't use the staticcheck.conf file reliably.

We didn't try initialisms like you proposed although that would be interesting if *initialisms* is the way over *checks*.

**Comment by ldez** [comment]
More explanation about staticcheck configuration in: https://github.com/golangci/golangci-lint/discussions/1989#discussioncomment-745004

Closed in favor of #1275
